packages/app/src/components/PageEditor/CodeMirrorEditor.jsx

```diff
@@ -665,16 +665,42 @@ export default class CodeMirrorEditor extends AbstractEditor {
     );
   }
 
-  closeEmojiPicker() {
-    this.setState({ isEmojiPickerShown: false });
+  getSearchCursor() {
+    const cm = this.getCodeMirror();
+    const pattern = /:[^:\s]+/;
+    const currentPos = cm.getCursor();
+    const sc = cm.getSearchCursor(pattern, currentPos, { multiline: false });
+    return sc;
+  }
+
+  addEmoji(emoji) {
+    const cm = this.getCodeMirror();
+    const currentPos = cm.getCursor();
+    const doc = cm.getDoc();
+    const sc = this.getSearchCursor();
+    if (sc.findPrevious() && this.state.isInputtingEmoji) {
+      sc.replace(emoji.colons, cm.getTokenAt(currentPos).string);
+      this.setState({ emojiSearchText: null });
+      cm.focus();
+    }
+    else {
+      doc.replaceRange(emoji.colons, currentPos);
+      cm.focus();
+    }
+
+  }
```

Comment:
Transplant to an utility module like EmojiAutoCompleteHelper please.

I think it's better not to write such emoji specific processes to CodeMirrorEditor component.